### PR TITLE
chore(webpack): update dependencies and use webpack

### PR DESCRIPTION
**Pull request by cbourgois**
## chore(webpack)

Remove `grunt` and use `webpack`

### Todo before merge : 

* [ ] clean dist generation to avoid package dependencies
* [ ] load translations
* [x] rebase on `master` 
* [ ] remove `UNIVERSE`

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      "jasmine": true,
+      "mocha": true,
+      "jquery": true,
+      "angular/mocks": true
+    },
+    "globals": {
+      "Raven": true
+    },
+    "rules": {
+      "no-underscore-dangle": 0
```
Comment (antleblanc) [inline .eslintrc]: Could you remove this? (also set `indent_size = 2`)

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    "plugins": ["angular"],
+    "env": {
+      "browser": true,
+      "jasmine": true,
+      "mocha": true,
+      "jquery": true,
+      "angular/mocks": true
+    },
+    "globals": {
+      "Raven": true
```
Comment (antleblanc) [inline .eslintrc]: Is it required?

File: .gitignore [hunk 3]
```diff
@@ -3,6 +3,5 @@ node_modules
 .tmp
 .idea
 npm-debug.log
-bower_components
```
Comment (antleblanc) [inline .gitignore]: I bet you can only keep:

```txt
node_modules
tools
```

File: src/otrs-popup/otrs-popup.less [hunk 4]
```diff
@@ -1,5 +1,13 @@
+@import '../node_modules/bootstrap/less/bootstrap';
+@import '../node_modules/font-awesome/less/font-awesome';
+
 @module-otrs-header-color: #3b4151;
 
+// @screen-md: 992px;
```
Comment (antleblanc) [inline src/otrs-popup/otrs-popup.less]: Could be removed?

Comment (cbourgois) [reply src/otrs-popup/otrs-popup.less]: of course

File: .eslintrc (hunk 2 above)
Comment (cbourgois) [reply .eslintrc]: wrong rebase :confused:

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        ],
+      },
+      {
+        test: /\.js$/,
+        exclude: /node_modules/,
+        use: [
+          {
+            loader: 'babel-loader',
+            options: {
+              presets: ['@babel/preset-env'],
```
Comment (antleblanc) [inline webpack.config.js]: Could be move to a `.babelrc` file.

File: package.json [hunk 6]
```diff
@@ -2,34 +2,54 @@
...
+    "eslint-config-airbnb-base": "^13.0.0",
+    "eslint-plugin-angular": "^3.3.0",
+    "eslint-plugin-import": "^2.13.0",
+    "file-loader": "^2.0.0",
     "jasmine-core": "^2.6.4",
-    "matchdep": "^0.3.0"
+    "less": "^2.3.1",
+    "less-loader": "^4.1.0",
+    "less-plugin-remcalc": "^0.0.1",
+    "matchdep": "^0.3.0",
```
Comment (antleblanc) [inline package.json]: I think you can safely remove `matchdep` from you `devDependencies`.

File: webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
+const webpack = require('webpack');
+const path = require('path');
+const ngAnnotatePlugin = require('ng-annotate-webpack-plugin');
+const RemcalcPlugin = require('less-plugin-remcalc');
+
+const config = {
+  mode: 'development',
```
Comment (antleblanc) [inline webpack.config.js]: `mode: 'production'` instead.

File: webpack.config.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+            loader: 'resolve-url-loader',
+          },
+          {
+            loader: 'less-loader', // compiles Less to CSS
+            options: {
+              sourceMap: true,
+              plugins: [
+                RemcalcPlugin,
+              ],
+              paths: [
```
Comment (antleblanc) [inline webpack.config.js]: Does the `paths` array is required? Otherwise you could use the `~`.

File: webpack.config.js [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+      $: 'jquery',
+      jQuery: 'jquery',
+      jquery: 'jquery',
+      'window.jQuery': 'jquery',
+    }),
+    new ngAnnotatePlugin({
+      add: true,
+      // other ng-annotate options here
+    }),
+  ],
```
Comment (antleblanc) [inline webpack.config.js]: You probably have to set:

```js
externals: {
  angular: 'angular',
},
```

**Comment by antleblanc** [comment]
`.npmignore` and `bower.json` files are no longer needed/up to date ;)

**Comment by jleveugle** [comment]
Hello @cbourgois,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Pull request closed without merging**